### PR TITLE
Implement step-definitions for login feature test

### PR DESCRIPTION
**Pull request by simrant27**
This PR implements step-definitions and page object model for login feature test.

### CODE REVIEW

File: tests/stepDefinitions/loginContext.js [hunk 1]
```diff
@@ -1,4 +1,36 @@
 const { Given, When, Then } = require("@cucumber/cucumber");
 const { expect } = require("@playwright/test");
+const loginBtnLocator = "input[value='Login']";
+const emailFieldLocator = "#spree_user_email";
+const passwordFieldLocator = "#spree_user_password";
+const alertLocator = "div.alert-danger"
 
-Given('the admin has navigated to the login page', function () {
+Given('user {string} has navigated to the admin login page', async function (string) {
```
Comment (prashant-gurung899) [inline tests/stepDefinitions/loginContext.js:8]: Name your variables that accurately describe the variable’s purpose, function, or the concept it represents.
```suggestion
Given('user {string} has navigated to the admin login page', async function (user) {
```

File: tests/stepDefinitions/loginContext.js [hunk 2]
```diff
@@ -1,4 +1,36 @@
...
+When('user {string} tries to log in with following credentials', async function (username, credentials) {
+    credentials = credentials.hashes();
+    for(let credential of credentials){
+        await page.locator(emailFieldLocator).fill(credential.email);
+        await page.locator(passwordFieldLocator).fill(credential.password);
+        await page.locator(loginBtnLocator).click();
+    }
+});
+
+Then('user {string} should be navigated to the admin panel dashboard', async function (username) {
```
Comment (prashant-gurung899) [inline tests/stepDefinitions/loginContext.js:29]:
```suggestion
Then('user {string} should be navigated to the admin panel dashboard', async function (user) {
```

File: tests/stepDefinitions/loginContext.js [hunk 3]
```diff
@@ -1,4 +1,36 @@
 const { Given, When, Then } = require("@cucumber/cucumber");
 const { expect } = require("@playwright/test");
+const loginBtnLocator = "input[value='Login']";
```
Comment (anon-pradip) [inline tests/stepDefinitions/loginContext.js:3]: IMHO, it is better to separate the related snippets like this:
```suggestion
const { expect } = require("@playwright/test");

const loginBtnLocator = "input[value='Login']";
```

File: tests/pageObjects/LoginPage.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    async enterCredentials(email,password){
+        await page.locator(this.emailFieldLocator).fill(email);
+        await page.fill(this.passwordFieldLocator, password);   
+    }
+
+    async clickLoginBtn(){
+        await page.locator(this.loginBtnLocator).click();
+    }
```
Comment (prashant-gurung899) [inline tests/pageObjects/LoginPage.js:20]: You can combine these functions into one
```suggestion
    async login(email,password){}
```

**Pull request merged**